flake.nix

```diff
@@ -19,76 +16,70 @@
 
     perSystem = { self', pkgs, lib, ... }:
       let
```

Comment:
```suggestion
    perSystem = { self', system, lib, ... }:
      let
        pkgs = import nixpkgs {
          inherit system;
          overlays = [ (import rust-overlay) ];
        };

```

Reply:
This is probably wrong and may not work. https://flake.parts/overlays#consuming-an-overlay gives a better example.

Reply:
Indeed, you're right, the example you pointed out is the preffered way to do that.

Reply:
Fenix has a crane example and is part of nix-community, we could consider using it. https://github.com/nix-community/fenix